src/systemdhook.c

```diff
@@ -378,59 +385,65 @@ static int move_mounts(const char *id,
 		       char *options
 	) {
 
+	/* Create the PATH directory */
+	if (contains_mount(id, config_mounts, config_mounts_len, path))
+		return 0;
+
 	char mount_dir[PATH_MAX];
 	snprintf(mount_dir, PATH_MAX, "%s%s", rootfs, path);
+	if (mkdir(mount_dir, 0755) == -1) {
```

Comment:
I don't see any cleanup for this if the function fails after this command succeeds.  Do you have a clever `_cleanup_umount_` analog for thise cases?  I usually approach this sort of thing with code like:

```c
  int rc = 0;
  char mount_dir[PATH_MAX] = "\0";
  ...
  snprintf(mount_dir, PATH_MAX, "%s%s", rootfs, path);
  if mkdir(mount_dir, 0755) == -1) {
    rc = -1;
    goto cleanup;
  }
  ...
 cleanup:
  if (rc != 0 && mount_dir[0] != '\0') {
    /* FIXME: unmount anything which had been mounted under mount_dir */
    if (umount(mount_dir) == -1) {
      perror("umount mount directory"); /* or however you're logging errno for failures */
    }
  }
  return rc;
}
```

Reply:
@wking I think your example above is flawed.  Did you want to rmdir(mount_dir)?  Your example shows you making a dir, but not mounting on it. 

We don't cleanup the mount_dir since it is on a tmpfs /run.

The cleanup_umount should umount the mountpoint if something fails before the MS_MOVE.